### PR TITLE
Live: max_connections option with strict default

### PR DESCRIPTION
**Pull request by FZambia**
This should help Live to be enabled by default but still do not affect setups with lots of simultaneous users. To properly handle many WS connections Grafana administrators should tune infrastructure a bit - at least increase a number of open files for a process. Will be in more details in documentation. Without this limit we risk to break all HTTP requests to Grafana server for setups with lots of active users (since all descriptors will be allocated for persistent connections).

### CODE REVIEW

File: public/app/features/live/LiveConnectionWarning.tsx [hunk 1]
```diff
@@ -45,8 +45,8 @@ export class LiveConnectionWarning extends PureComponent<Props, State> {
   render() {
     const { show } = this.state;
     if (show) {
-      if (!contextSrv.isSignedIn) {
-        return null; // do not show the warning for anonomous users (and /login page etc)
+      if (!contextSrv.isSignedIn || !config.liveEnabled) {
```
Comment (ryantxu) [inline public/app/features/live/LiveConnectionWarning.tsx:48]: This only shows up in dev mode, so it is not urgent...

BUT we may also want to check the disconnect reason to know if we should really show an error or not

File: conf/defaults.ini [hunk 2]
```diff
@@ -887,6 +887,12 @@ allow_loading_unsigned_plugins =
 plugin_admin_enabled = false
 plugin_catalog_url = https://grafana.com/grafana/plugins/
 
+#################################### Grafana Live ##########################################
+[live]
+# max_connections to Grafana Live WebSocket endpoint per Grafana server instance. See Grafana Live docs if you are
+# going to make it higher than default 100 since this can require some OS and infrastructure tuning. -1 for unlimited.
```
Comment (ryantxu) [inline conf/defaults.ini:893]: we should also say that 0 will disable clients connecting?

File: pkg/services/live/live.go [hunk 3]
```diff
@@ -177,6 +177,15 @@ func (g *GrafanaLive) Init() error {
 	// different goroutines (belonging to different client connections). This is also
 	// true for other event handlers.
 	node.OnConnect(func(client *centrifuge.Client) {
+		numConnections := g.node.Hub().NumClients()
+		if g.Cfg.LiveMaxConnections >= 0 && numConnections >= g.Cfg.LiveMaxConnections+1 {
```
Comment (FZambia) [inline pkg/services/live/live.go:181]: numConnections > g.Cfg.LiveMaxConnections

**Review by ryantxu — APPROVED** [review]
looks good -- but I have not stress tested this to know that it actually works :)

File: conf/defaults.ini (hunk 2 above)
Comment (FZambia) [reply conf/defaults.ini]: done

**Pull request merged**